includes/class-wpcom-legacy-redirector-ui.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+class WPCOM_Legacy_Redirector_UI {
+
+    public function admin_menu() {
+        add_menu_page( 'WPCOM Legacy Redirector', 'Redirect Manager', 'manage_options', 'wpcom-legacy-redirector', array(
+            $this,
+            'generate_page_html',
+        ), 'dashicons-external' );
+    }
+
+    public function generate_page_html() {
+        if ( ! current_user_can( 'manage_options' ) ) {
+            return;
+        }
+
+        $errors   = array();
+        $messages = array();
+        if ( class_exists( 'WPCOM_Legacy_Redirector' ) ) {
+            if ( isset( $_POST['from_url'] ) && isset( $_POST['redirect_to'] ) ) {
+                if (
+                    ! isset( $_POST['redirect_nonce_field'] )
+                    || ! wp_verify_nonce( $_POST['redirect_nonce_field'] )
+                ) {
+                    $errors[] = array(
+                        'label'   => __( 'Error', 'wpcom-legacy-redirector' ),
+                        'message' => __( 'Sorry, your nonce did not verify.', 'wpcom-legacy-redirector' ),
+                    );
+                } else {
+                    $from_url    = sanitize_text_field( $_POST['from_url'] );
+                    $redirect_to = sanitize_text_field( $_POST['redirect_to'] );
```

Comment:
well the thing about this plugin is nothing happens if there's not a 404 (ie: the lookup happens only when a 404 is detected.)  But this isn't a vital improvement -- right now you can add urls that do not 404, and therefore will never actually be redirected -- fixing this issue (if it can be fixed) is probably outside of the scope here.